### ISSUE TITLE
When 'spring-boot-devtools' is introduced in the project, the extensions are not available.

### ISSUE THREAD

**Issue by octoape** [pf4j/pf4j-spring#77]
Hello @decebals

When 'spring-boot-devtools' is introduced in the project, the extensions are not available. If I remove 'spring-boot-devtools', I can get the extensions normally. Is there a way to solve this problem? 'spring-boot-devtools' is very useful when developing, so I can't just delete 'spring-boot-devtools'.

Thanks.

**Comment by decebals** [comment]
I don't know what to say about this problem. I don't use spring-boot in my projects. Try to investigate/debug the problem.

**Comment by decebals** [comment]
For SpringBoot and PF4J integration please take a look on https://github.com/hank-cp/sbp.

**Comment by decebals** [comment]
From what I see in the [Spring DevTools reference page]( https://docs.spring.io/spring-boot/reference/using/devtools.html), it's a caution related to this aspect.

> Devtools might cause classloading issues, in particular in multi-module projects. [Diagnosing Classloading Issues](https://docs.spring.io/spring-boot/reference/using/devtools.html#using.devtools.diagnosing-classloading-issues) explains how to diagnose and solve them.

**Comment by decebals** [comment]
First step is to try to follow some recommendations described in the [Diagnosing Classloading Issues](https://docs.spring.io/spring-boot/reference/using/devtools.html#using.devtools.diagnosing-classloading-issues) section.

**Comment by grimsi** [comment]
I was able to get it working with Spring Devtools restart functionality enabled thanks to stefv from StackOverflow: https://stackoverflow.com/questions/73654174/my-application-cant-find-the-extension-with-pf4j

See this commit: https://github.com/gameyfin/gameyfin/commit/0563f2b36695abe4cf5a64fbb6d17e565a06377b

TL;DR: I extended the PluginManager with a custom PluginLoader which gets its ClassLoader injected from the Spring Context.
Maybe this can be included in PF4J-Spring out of the box? Seems like a fairly common case that Devtools are used when Spring (Boot) is used.

**Comment by decebals** [comment]
> Maybe this can be included in PF4J-Spring out of the box? Seems like a fairly common case that Devtools are used when Spring (Boot) is used.

You can create a PR.
